test/models/work_test.rb

```diff
@@ -0,0 +1,166 @@
+require "test_helper"
+
+describe Work do
+
+  describe "validations" do
+    it "is valid when all fields are present" do
+      work = Work.new(category: "book", title: "Big Friendship", publication_year: "2020", description: "A book about friendship")
+      result = work.valid?
+      expect(result).must_equal true
+    end
+
+    it "must have a category" do
+      result = works(:hp1)
+      result[:category] = nil
+      expect(result.valid?).must_equal false
+    end
+
+    it "must have a title" do
+      result = works(:hp1)
+      result[:title] = nil
+      expect(result.valid?).must_equal false
+    end
+
+    it "must have a unique title" do
+      book = works(:hp1)
+      book_copy = Work.new(category: "book", title: book.title)
+      expect(book_copy.valid?).must_equal false
+    end
+  end
+
+  describe "custom methods" do
+
+    before do
+
+      Vote.all.each do |vote|
+        vote.delete
+      end
+      # The counter cache doesn't get updated for fixtures, have to manually create votes
```

Comment:
Interesting! I would be curious to hear more about this. I'm not sure what counter cache means but I'm glad you were able to find a way around it, even though it looks like it created quite a bit of extra work for you :/